configserver/src/main/java/com/yahoo/vespa/config/server/application/DefaultClusterReindexingStatusClient.java

```diff
@@ -0,0 +1,150 @@
+
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final Executor executor =
+            Executors.newSingleThreadExecutor(new DaemonThreadFactory("cluster-controller-reindexing-client-"));
+    private final CloseableHttpAsyncClient httpClient = createHttpClient();
+
+    public DefaultClusterReindexingStatusClient() {
+        httpClient.start();
+    }
+
+    @Override
+    public Map<String, ClusterReindexing> getReindexingStatus(ModelResult application) throws IOException {
+        Map<ClusterId, List<ServiceInfo>> clusters = clusterControllerClusters(application);
+        Map<ClusterId, CompletableFuture<ClusterReindexing>> futureStatusPerCluster = new HashMap<>();
+        clusters.forEach((clusterId, clusterNodes) -> {
+            var parallelRequests = clusterNodes.stream()
+                    .map(this::getReindexingStatus)
+                    .collect(Collectors.toList());
+            CompletableFuture<ClusterReindexing> combinedRequest = CompletableFutures.firstOf(parallelRequests);
+            futureStatusPerCluster.put(clusterId, combinedRequest);
+        });
+
+        try {
+            Map<String, ClusterReindexing> statusPerCluster = new HashMap<>();
+            futureStatusPerCluster.forEach((clusterId, futureStatus) -> {
+                statusPerCluster.put(clusterId.s(), futureStatus.join());
+            });
+            return statusPerCluster;
+        } catch (Exception e) {
+            throw new IOException("Failed to get reindexing status from cluster controllers: " + e.getMessage(), e);
+        }
+    }
+
+    @Override public void close() { uncheck(() -> httpClient.close()); }
+
+    private CompletableFuture<ClusterReindexing> getReindexingStatus(ServiceInfo service) {
+        URI uri = URI.create(String.format("http://%s:%d/reindexing/v1/status", service.getHostName(), getStatePort(service)));
+        CompletableFuture<SimpleHttpResponse> responsePromise = new CompletableFuture<>();
+        httpClient.execute(SimpleHttpRequests.get(uri), new FutureCallback<>() {
+            @Override public void completed(SimpleHttpResponse result) { responsePromise.complete(result); }
+            @Override public void failed(Exception ex) { responsePromise.completeExceptionally(ex); }
+            @Override public void cancelled() { responsePromise.cancel(false); }
+        });
+        return responsePromise.handleAsync((response, error) ->  {
+            if (response != null) {
+                return uncheck(() -> toClusterReindexing(response));
+            } else {
+                throw throwUnchecked(new IOException(String.format("For '%s': %s", uri, error.getMessage()), error));
+            }
+        }, executor);
+    }
+
+    private static ClusterReindexing toClusterReindexing(SimpleHttpResponse response) throws IOException {
+        if (response.getCode() != HttpStatus.SC_OK) throw new IOException("Expected status code 200, got " + response.getCode());
```

Comment:
Perhaps we could return status from first successful, if the first to return a response is non-200?